app/models/miq_provision_virt_workflow.rb

```diff
@@ -1077,7 +1077,9 @@ def get_selected_hosts(src)
             elsif src[:host_id]
               selected_host(src)
             else
-              load_ar_obj(allowed_hosts)
+              allowed_hosts.group_by(&:evm_object_class).flat_map do |type, objs|
```

Comment:
@NickLaMuro - do you think there might be a way to add the `group_by` behavior into `load_ar_obj`?

Reply:
@syncrou Yeah, I was thinking about that.  I mostly decided to shy away from that approach since isolating the code to this method would avoid potential risk of an unknown edge case of `load_ar_obj` that I was unware of.

That said, if you would prefer me trying to make this code work there, I can see what I can do.

Reply:
@NickLaMuro - I would.  Largely because that method is already abstracting out other potential edge cases.  It should be easier to test the changes when applied to that method as well.

Reply:
Since this go merged, I am going to address this specific case separately in another PR.